components/dialogs/AuthDialog.vue

```diff
@@ -318,25 +330,110 @@ export default {
       switch (platform) {
         case 'email':
           this.currentTab = 'email';
-          break;
+          return;
 
         case 'wallet':
           this.currentTab = 'loading';
           this.startWeb3AndCb(this.signInWithWallet);
-          break;
+          return;
 
-        default: {
+        case 'google':
+        case 'twitter':
           this.signInPayload = await firebasePlatformSignIn(platform);
-          this.login();
-        }
+          break;
+
+        case 'facebook':
+          try {
+            this.currentTab = 'loading';
```

Comment:
make it a method `setCurrentTabLoading()` sounds more robust?

Reply:
Probably change it later